packages/valibot-validator/src/index.ts

```diff
@@ -56,6 +56,6 @@ export const vValidator = <
       return c.json(result, 400)
```

Comment:
@yusukebe is there an advantage to returning `result` instead of `result.issues`?

Reply:
Hi @fabian-hiller 

I think `result` is not bad. And if they do not prefer it, they can change it themselves with the hook.

@chimame Do you have any thoughts?

Reply:
I understand that the intention of fabian-hiller's comment is as follows.

If validation fails, wouldn't it be fine to set the HTTP status to 400 and return only the error details in JSON? I think that is what he is trying to suggest.

If validation is successful, you will get `result.output`, in other words, the JSON that passed validation. However, if validation fails, the entire `result` JSON will be returned to the client. The type of `result` is as follows. The type has been simplified for ease of understanding.

```ts
type SafeParseResult = 
  | {
      readonly typed: true;
      readonly success: true;
      readonly output: InferOutput<TSchema>; // <-- Contains objects that have been successfully validated.
      readonly issues: undefined; // <-- Since the validation was successful, there are no issues.
    }
  | {
      readonly typed: false;
      readonly success: false;
      readonly output: unknown; // <-- Contains the original data that failed validation.
      readonly issues: [InferIssue<TSchema>, ...InferIssue<TSchema>[]]; // <-- This contains the details of the errors for each item that failed validation.
    };

const result: SafeParseResult = await safeParseAsync(schema, value);

/*
example error result
{
  typed: false,
  success: false,
  output: { name: 'Superman', age: '20' },
  issues: [
    {
      kind: 'schema',
      type: 'number',
      input: '20',
      expected: 'number',
      received: '"20"',
      message: 'Invalid type: Expected number but received "20"',
      requirement: undefined,
      path: [Array],
      issues: undefined,
      lang: undefined,
      abortEarly: undefined,
      abortPipeEarly: undefined,
      skipPipe: undefined
    }
  ]
}
*/
```

If that is the intention, it is certainly not a bad idea to only return issues.

```suggestion
      return c.json({ issues: result.issues }, 400)
```

Reply:
In general `result` is not bad and can be better because it contains more info, but if this info is never used it is easier for the user to just get the issues.

```ts
return c.json(result.issues, 400)
```

I haven't used Hono with Valibot, so I can't say. It was just a question that came to mind while reviewing the code. I saw that the Zod adapter also returns `result`. We should keep it consistent.

Reply:
> In general `result` is not bad and can be better because it contains more info, but if this info is never used it is easier for the user to just get the issues.

Yeah, I can understand both reasons. When I created the Zod Validator, I used `result` because I thought more information was better. Shall we go with `result` as it is? Then, we will not have a breaking change.

Thanks for the comments!

Reply:
Yes, let's leave it for now. 👍🏼